modules/ossm-uguide-routing-ingress.adoc

```diff
@@ -0,0 +1,228 @@
+----
+
+----
+$ export INGRESS_PORT=$(oc -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="http2")].port}')
+----
+
+----
+$ export SECURE_INGRESS_PORT=$(oc -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="https")].port}')
+----
+
+In some environments, the load balancer may be exposed using a host name instead of an IP address. For that case, the ingress gateway's `EXTERNAL-IP` value is not be an IP address. Instead, it's a host name, and the previous command fails to set the `INGRESS_HOST` environment variable.
+
+Use the following command to correct the `INGRESS_HOST` value:
+
+----
+$ export INGRESS_HOST=$(oc -n istio-system get service istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].hostname}')
+----
+
+.Node port
+
+Follow these instructions if your environment does not have an external load balancer. You must use a node port instead.
+
+Set the ingress ports:
+
+----
+$ export INGRESS_PORT=$(oc -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="http2")].nodePort}')
+----
+
+----
+$ export SECURE_INGRESS_PORT=$(oc -n istio-system get service istio-ingressgateway -o jsonpath='{.spec.ports[?(@.name=="https")].nodePort}')
+----
+
+Setting the ingress IP depends on the cluster provider:
+
+*  _GKE:_
+
+    $ export INGRESS_HOST=<workerNodeAddress>
+
+    You must create firewall rules to allow the TCP traffic to the _ingressgateway_ service's ports. Run the following commands to allow the traffic for the HTTP port, the secure port (HTTPS) or both:
+
+    $ gcloud compute firewall-rules create allow-gateway-http --allow tcp:$INGRESS_PORT 
+
+    $ gcloud compute firewall-rules create allow-gateway-https --allow tcp:$SECURE_INGRESS_PORT
++
+*  _Minikube:_
+
+    $ export INGRESS_HOST=$(minikube ip)
++
+*  _Docker For Desktop:_
+
+    $ export INGRESS_HOST=127.0.0.1
+
++
+*  _Other environments, such as IBM Cloud Private_:
+
+    $ export INGRESS_HOST=$(oc get po -l istio=ingressgateway -n istio-system -o jsonpath='{.items[0].status.hostIP}')
+
+[id="ossm-ingress-config-_{context}"]
+== Configuring ingress using an {ProductShortName} gateway
+
+An ingress gateway describes a load balancer operating at the edge of the mesh that receives incoming HTTP/TCP connections.
+
+It configures exposed ports and protocols, but does not include any traffic routing configuration. Traffic routing for ingress traffic is configured with {ProductShortName} routing rules, which is the same way internal service requests are configured.
+
+Configure a gateway on port 80 for HTTP traffic.
+
+1.  Create an {ProductShortName} `Gateway`:
+
+    $ oc apply -f - <<EOF
+    apiVersion: networking.istio.io/v1alpha3
+    kind: Gateway
+    metadata:
+      name: httpbin-gateway
+    spec:
+      selector:
+        istio: ingressgateway # use {ProductShortName} default gateway implementation
```

Comment:
This `{ProductShortName}` is not expanding, it's being rendered as raw `{ProductShortName}`. You can just cut it off.